internal/resources/kubernetessecret/resource_secret.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package kubernetessecret
+
+import (
+	"github.com/go-openapi/strfmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	secretmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/kubernetessecret/cluster"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/kubernetessecret/scope"
+)
+
+func ResourceSecret() *schema.Resource {
+	return &schema.Resource{
+		Schema: secretSchema,
+	}
+}
+
+var secretSchema = map[string]*schema.Schema{
+	NameKey: {
+		Type:     schema.TypeString,
+		Required: true,
+		ForceNew: true,
+	},
+	NamespaceNameKey: {
+		Type:     schema.TypeString,
+		Required: true,
+		ForceNew: true,
+	},
+	OrgIdKey: {
+		Type: schema.TypeString,
+	},
+	scope.ScopeKey: scope.ScopeSchema,
+	specKey:        secretSpec,
+	ExportKey: {
+		Type:     schema.TypeBool,
+		Required: true,
+	},
+	statusKey: {
+		Type:     schema.TypeMap,
+		Computed: true,
+		Elem:     &schema.Schema{Type: schema.TypeString},
+	},
+}
+
+var secretSpec = &schema.Schema{
+	Type:        schema.TypeList,
+	Description: "Spec for the kubernetes secret",
+	Optional:    true,
+	MaxItems:    1,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			SecretTypeKey: {
+				Type:        schema.TypeString,
+				Description: "SecretType definition - indicates the kubernetes secret type.",
```

Comment:
presently we are supporting only one type so mention that in the description and even the value

Reply:
updated PTAL